scalafix-rules/src/main/scala/scalafix/internal/rule/ExplicitResultTypes.scala

```diff
@@ -173,13 +173,21 @@ final class ExplicitResultTypes(
     def hasParentWihTemplate: Boolean =
       defn.parent.exists(_.is[Template])
 
-    isImplicit && !isFinalLiteralVal || {
+    def isImplicitAndNotFinalLiteralVal: Boolean =
+      isImplicit && !isFinalLiteralVal
+
+    def matchesMemberKindAndVisibility: Boolean =
+      matchesMemberKind && matchesMemberVisibility
+
+    def evaluateWhenNotImplicitOrFinalLiteralVal: Boolean = {
```

Comment:
if you have a better name for this block it let me know, I couldn't find a better one not based on the first `isImplicitAndNotFinalLiteralVal` block

Reply:
the name doesn't reflect what the method does (or the content of the method) but reflects which branch of the logical expression of isRuleCandidate is executed, in relation with the first block in OR. This makes the logical expression more readable but the method itself less readable..it's a trade off and I think it's better if the logical expression is more readable, that's why we extracted the method in the first place